### DISCUSSION TITLE
Couldn't load custom interface

### DISCUSSION THREAD

**Discussion by sandyksit** [directus/directus#8780] [Archive]
I am using new version 9, but
Failed to load module script: Expected a JavaScript module script but the server responded with a MIME type of "application/json".Strict MIME type checking is enforced for module scripts per HTML type.
Couldn't load custom interface "button-preview": TypeError: failed to fetch dynamically imported module :http://domain.com:8085/extentions/interfaces/button-preview/index.js

folder structure like:
extensions>button-preview>src
for build and deploy using `npx rollup -c`

**Comment by joselcvarela** [comment]
Hello @sandyksit 
I don't know if this is the case, but the documentation says: 
https://docs.directus.io/guides/interfaces/#_4-build-and-deploy

So just to be sure, are you copying the content of `dist` folder into the `button-preview` folder?
The equivalent command  should be `mv dist extensions/interfaces/button-preview`

**Reply by sandyksit** [reply]
Hello @joselcvarela 
after run the command `npx rollup -c`
dist folder available in the button-preview folder
folder structure is look like this 
`  

- extenstions
    -interfaces

       -button-preview
           -src
              - index.js
               -interface.vue
           -dist
           -node_modules
           -rollup.config.js
           -package.json
           -pacakge-lock.json    `

could you please confirm its right way?

**Reply by joselcvarela** [reply]
This is what I was afraid of 😋
You should move `dist/index.js` to `button-preview`.
Maybe you could put the content you have right now in `button-preview` inside `button-preview/development` folder for example, just to be more clean

In the end you should have something like this:
```
extensions
  interfaces
    button-preview
      development
      index.js <-- from development/dist/index.js

**Reply by sandyksit** [reply]
![200DA8D8-7CEA-4DA5-9B51-4336CE6D7B70](https://user-images.githubusercontent.com/13213679/137272574-7a5a6f7e-28f3-44a9-b818-2bcf60c4bbda.jpeg)

I am trying same, but I am not able to see in UI

**Reply by joselcvarela** [reply]
On starting starting Directus (`npx directus start`) you should see `Loaded extensions: button-preview`. Are you seeing it?
Then, your custom interface will only appear for the specific type you have defined, for example:
```
import InterfaceComponent from './interface.vue';

export default {
	id: 'custom',
	name: 'Custom',
	description: 'This is my custom interface!',
	icon: 'box',
	component: InterfaceComponent,
	types: ['string'], // <- this will only appear on string fields
};
```

**Reply by sandyksit** [reply]
It’s displaying Couldn’t load custom-interface

**Reply by joselcvarela** [reply]
Try this:
- clear folder `button-preview` (move everything to another folder for example)
- create a file `index.js` inside `button-preview`
- put this content inside file `index.js`
```
import{openBlock as e,createElementBlock as t}from"vue";var n={emits:["input"],props:{value:String},methods:{handleChange(e){this.$emit("input",e)}}};const i=["value"];n.render=function(n,u,a,o,r,s){return e(),t("input",{value:a.value,onInput:u[0]||(u[0]=e=>s.handleChange(e.target.value))},null,40,i)},n.__file="interface.vue";var u={id:"custom",name:"Custom",description:"This is my custom interface!",icon:"box",component:n,types:["string"]};export{u as default};
```
- check if interface appears for `string` fields

**Reply by sandyksit** [reply]
![image](https://user-images.githubusercontent.com/13213679/137272252-1018d9f8-3e48-4b48-97d2-a170f119c9af.jpg)

I am following same .
Create folder button-preview inside interface ,
Create folder development inside button-preview, 
Create package.json and rollup.config.js
And two file index.js and interface.vue inside development folder,
Create index.js file inside button-preview folder and pasted your given string.
But still not working

**Reply by nickrum** [reply]
Which RC version of Directus are you using? Looking at those error messages, I think your version is quite old.

**Reply by sandyksit** [reply]
I am using version 9.0.0.rc.67

**Reply by sandyksit** [reply]
Now i have updated the version with 9.0.0.rc.98, now i am not seeing any error on load, but my custom interface is not displaying.

**Reply by joselcvarela** [reply]
Can you show the content of the file `extensions/interfaces/button-preview/index.js` ?

**Reply by sandyksit** [reply]
`import{openBlock as e,createElementBlock as t}from"vue";var n={props:{value:String},methods:{handleChange(e){this.$emit("input",e)}}};const a=["value"];n.render=function(n,i,r,u,o,s){return e(),t("input",{value:r.value,onInput:i[0]||(i[0]=e=>s.handleChange(e.target.value))},null,40,a)},n.__file="src/interface.vue";var i={id:"custom",name:"Custom",description:"This is my custom interface!",icon:"box",component:n,types:["string"]};export{i as default};
`

**Reply by joselcvarela** [reply]
I have tested your file here locally and seems to have no problems:
<img src="https://user-images.githubusercontent.com/14039341/138032255-02e7ac6c-e371-476a-a660-86397d8c533e.png" width="400">

**Reply by sandyksit** [reply]
Thanks, I am also able to see that

**Reply by sandyksit** [reply]
I am trying to create custom preview button, but it's creating input box, could you please guide me, its right or not?
`import{openBlock as e,createElementBlock as n}from"vue";var o={props:{value:String},methods:{handleChange(e){console.log("clicked",e)}}};o.render=function(o,r,t,i,c,s){return e(),n("button",{onClick:r[0]||(r[0]=e=>s.handleChange(e))},"Preview")},o.__file="src/interface.vue";var r={id:"custom",name:"Custom",description:"This is my custom interface for preview!",icon:"box",component:o,types:["string"]};export{r as default};
`

**Reply by sandyksit** [reply]
@joselcvarela please help me why button is not displaying? And how refresh the existing interface? I have change the interface its not reflecting new one

**Reply by joselcvarela** [reply]
Now is up to you develop your custom interface https://docs.directus.io/guides/interfaces/#custom-interfaces
Right now you have an input because this was on the example:
```vue
// src/interface/vue.js
<template>
	<input :value="value" @input="handleChange($event.target.value)" /> <!-- You need to customize this with your own code-->
</template>

<script>
export default {
	emits: ['input'],
	props: {
		value: String,
	},
	methods: {
		handleChange(value) {
			this.$emit('input', value);
		},
	},
};
</script>
```
Then you need to run `npx directus-extension build` to generate the `dist/index.js` and finally you should put this file in `extensions/interfaces/button-preview/`

**Reply by sandyksit** [reply]
I have tried Now i have button but its not refreshing in the UI even custom description is also not updating, its displaying old one
`import{openBlock as e,createElementBlock as o}from"vue";var i={props:{value:String},methods:{handleClick(e){console.log("clicked",e)}}};i.render=function(i,r,n,t,c,s){return e(),o("button",{onClick:r[0]||(r[0]=e=>s.handleClick(e))},"Preview")},i.__file="src/interface.vue";var r={id:"custom",name:"Custom",description:"This is my custom interface for preview!",icon:"box",component:i,types:["string"]};export{r as default};`

**Reply by sandyksit** [reply]
@joselcvarela please help me why button is not displaying even i used button control?

**Reply by joselcvarela** [reply]
Hello @sandyksit
Run `npx directus-extension build -w`. This will generate new `dist/index.js`. Then copy the `index.js` into `extensions/interfaces/button-preview`.
The file destination of `npx directus-extension build -w` is configurable inside `package.json`, so you don't need to do the copy file.

**Reply by sandyksit** [reply]
I have follow the same steps, but its displaying old one, you can see above compiled code is 'button' code

**Reply by joselcvarela** [reply]
Try in anonymous tab. Some users experienced cache of extensions on client side.

Also, you can join our Discord server to get further help 
https://directus.chat

**Reply by sandyksit** [reply]
@joselcvarela  I tried many times and multiple anonymous tabs but still not refreshing the content

**Reply by azrikahar** [reply]
@sandyksit it would definitely be better to forward your question to the Discord server as @joselcvarela pointed out, since I firmly believe your current issue will benefit more from the easier back-and-forth in a chat environment, rather than a thread/comment natured setting. Hope that's understandable! 👍

**Reply by sandyksit** [reply]
I am able to get latest change, Problem is that its not reflecting automatically, latest changes is reflecting after stop and restart the server, But  this is not good. Thanks @joselcvarela
